src/nfc_reader/nfc_reader.py

```diff
@@ -1,12 +1,108 @@
-import nfc
+import json
+import os
+import logging
+import logging.config
+import time
+
+from src.nfc_reader.poll import poll
+from src.cloud_db.cloud_db import CloudDBClient
+from src.utils.paths import (NFC_READER_TO_SCREEN_PIPE, PROJECT_DIR)
+from src.utils.utils import get_device_id, TapStatus
+
+logging.config.fileConfig(f"{PROJECT_DIR}/config/logging.conf", disable_existing_loggers=False)
+logger = logging.getLogger(os.path.basename(__file__).split('.')[0])
+
+class NFCReader:
+    """Represents a NFC Reader.
+
+    Attributes:
+        debounce_time: Minimum time interval (in seconds) to consider consecutive taps as distinct.
+        screen_pipe: File path to the FIFO used for IPC with the screen module.
+        cloud_db: An instance of CloudDBClient for database interactions.
+    """
+
+    def __init__(self,
+                 debounce_time: int = DEBOUNCE_TIME,
+                 screen_pipe: str = NFC_READER_TO_SCREEN_PIPE) -> None:
+        """
+        Initializes the NFC Reader with specified debounce time and pipe path.
+
+        Args:
+            debounce_time: An integer specifying the debounce time in seconds.
+            screen_pipe: File path to the screen FIFO.
+        """
+        self.debounce_time = debounce_time
+        self.screen_pipe = screen_pipe
+        self.cloud_db = CloudDBClient()
+        self.last_tap = 0.0
+
+
+    def handle_tap(self, tag_id) -> bool:
```

Comment:
For NFC taps, I expect the behavior to be that there are only ever valid taps. The reason being is that there will be infrequent taps, so just registering them whenever you are open to it is good. The way I would use debounce time is the following: 
1. Poll and register your first NFC tap (utilizes poll(uid_str, uid_len))
2. Immediately check if a tag is present and wait until there is no tag present (utilizes is_tag_present())
3. Sleep for a short window (i.e. 250 ms)
4. Poll again

Reference `wrapper_test.py` in `feature/nfc-reader`

Reply:
Remove anything that is not needed as a consequence of these changes. For instance, the TapStatus is not needed.